pkg/cloudmap/watcher.go

```diff
@@ -140,7 +140,7 @@ func (w *watcher) hostsForNamespace(ctx context.Context, ns *sdTypes.NamespaceSu
 	return hosts, nil
 }
 
-func (w *watcher) endpointsForService(ctx context.Context, svc *sdTypes.ServiceSummary, ns *sdTypes.NamespaceSummary) ([]*v1alpha3.ServiceEntry_Endpoint, error) {
+func (w *watcher) endpointsForService(ctx context.Context, svc *sdTypes.ServiceSummary, ns *sdTypes.NamespaceSummary) ([]*v1alpha3.WorkloadEntry, error) {
```

Comment:
In general I would suggest to review all method names from all changed method's return types.
There are more like: `instancesToEndpoints` ==> `instancesToWorkloadEntries`
Finally, all the places were you call this methods that changed their return types, double check that the variable name matches with the accepted "Type". For instance (pun intended):
```
// ep := instanceToEndpoint(&inst)
ep := instanceToWorkloadEntry(&inst) // Wrong
wle := instanceToWorkloadEntry(&inst) // Better
```

```suggestion
func (w *watcher) workloadEntriesForService(ctx context.Context, svc *sdTypes.ServiceSummary, ns *sdTypes.NamespaceSummary) ([]*v1alpha3.WorkloadEntry, error) {
```